_includes/v22.1/sql/sql-examples.md

```diff
@@ -0,0 +1,593 @@
+~~~ shell
+$ cockroach-sql --insecure \
+--execute="CREATE TABLE t1 (id INT PRIMARY KEY, name STRING)" \
+--execute="INSERT INTO t1 VALUES (1, 'a'), (2, 'b'), (3, 'c')" \
+--user=maxroach \
+--host=12.345.67.89 \
+--database=db1
+--echo-sql
+~~~
+
+~~~
+# Server version: CockroachDB CCL f8f3c9317 (darwin amd64, built 2017/09/13 15:05:35, go1.8) (same version as client)
+# Cluster ID: 847a4ba5-c78a-465a-b1a0-59fae3aab520
+> SET sql_safe_updates = TRUE
+> CREATE TABLE t1 (id INT PRIMARY KEY, name STRING)
+CREATE TABLE
+> INSERT INTO t1 VALUES (1, 'a'), (2, 'b'), (3, 'c')
+INSERT 3
+~~~
+
+In this example, we start the interactive SQL shell and enable the `echo` shell option to reveal SQL statements sent implicitly:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ cockroach-sql --insecure \
+--user=maxroach \
+--host=12.345.67.89 \
+--database=db1
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+> \set echo
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+> INSERT INTO db1.t1 VALUES (4, 'd'), (5, 'e'), (6, 'f');
+~~~
+
+~~~
+> INSERT INTO db1.t1 VALUES (4, 'd'), (5, 'e'), (6, 'f');
+INSERT 3
+
+Time: 2.426534ms
+
+> SHOW TRANSACTION STATUS
+> SHOW DATABASE
+~~~
+
+### Repeat a SQL statement
+
+Repeating SQL queries on a table can be useful for monitoring purposes. With the `--watch` flag, you can repeat the statements specified with a `--execute` or `-e` flag periodically, until a SQL error occurs or the process is terminated.
+
+For example, if you want to monitor the number of queries running on the current node, you can use `cockroach-sql` with the `--watch` flag to query the node's `crdb_internal.node_statement_statistics` table for the query count:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ cockroach-sql --insecure \
+--execute="SELECT SUM(count) FROM crdb_internal.node_statement_statistics" \
+--watch 1m
+~~~
+
+~~~
+  sum
+-------
+  926
+(1 row)
+  sum
+--------
+  4227
+(1 row)
+^C
+~~~
+
+In this example, the statement is executed every minute. We let the process run for a couple minutes before terminating it with Ctrl+C.
+
+### Connect to a cluster listening for Unix domain socket connections
+
+To connect to a cluster that is running on the same machine as your client and is listening for [Unix domain socket](https://en.wikipedia.org/wiki/Unix_domain_socket) connections, [specify a Unix domain socket URI](connection-parameters.html#example-uri-for-a-unix-domain-socket) with the `--url` connection parameter.
+
+For example, suppose you start a single-node cluster with the following [`cockroach start-single-node`](cockroach-start-single-node.html) command:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ cockroach start-single-node --insecure --socket-dir=/tmp
+~~~
+
+~~~
+CockroachDB node starting at 2020-10-12 04:02:54.971369 +0000 UTC (took 1.3s)
+build:               CCL v20.2.0 @ 2020/10/06 17:15:06 (go1.13.14)
+webui:               http://Jesses-MBP-2:8080
```

Comment:
I'll assume that any reference to Jesse can be kept as an homage.

Reply:
> > ```gfm
> > ~~~
> > 
> > ### Edit SQL statements in an external editor
> > ```
> 
> Please do not include this section in the doc, as we're going to remove this functionality imminently. ([cockroachdb/cockroach#86457](https://github.com/cockroachdb/cockroach/pull/86457))

Removed.

> However, separately from this doc project, I'd like to hear your opinion on this: is the ability to use an external editor desirable? Do you think users will be grateful if we streamline this a little bit?

I think the usability of the internal editor should be addressed first. If users are breaking out of the shell to edit their SQL externally there are likely two reasons: 1) they find it too cumbersome to use the shell or 2) they are editor absolutists. We can't do much about 2), but we can address the usability of the shell for group 1). Syntax highlighting, and code completion would go a long way to making it easier to accomplish their tasks without needing an external editor.

> missing newline at end of file

Done.

> The output of `cockroach sql` _when used non-interactively_ is part of ...

Done.